airflow/providers/google/cloud/hooks/new_dataproc.py

```diff
@@ -0,0 +1,892 @@
+        """Returns JobControllerClient."""
+        client_options = {'api_endpoint': f'{location}-dataproc.googleapis.com:443'} if location else None
+
+        return JobControllerClient(
+            credentials=self._get_credentials(), client_info=self.client_info, client_options=client_options
+        )
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def create_cluster(
+        self,
+        region: str,
+        project_id: str,
+        cluster_name: str,
+        cluster_config: Union[Dict, Cluster],
+        labels: Optional[Dict[str, str]] = None,
+        # request_id: Optional[str] = None,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = None,
+    ):
+        """
+        Creates a cluster in a project.
+
+        :param project_id: Required. The ID of the Google Cloud project that the cluster belongs to.
+        :type project_id: str
+        :param region: Required. The Cloud Dataproc region in which to handle the request.
+        :type region: str
+        :param cluster_name: Name of the cluster to create
+        :type cluster_name: str
+        :param labels: Labels that will be assigned to created cluster
+        :type labels: Dict[str, str]
+        :param cluster_config: Required. The cluster config to create.
+            If a dict is provided, it must be of the same form as the protobuf message
+            :class:`~google.cloud.dataproc_v1.types.ClusterConfig`
+        :type cluster_config: Union[Dict, google.cloud.dataproc_v1.types.ClusterConfig]
+        :param request_id: Optional. A unique id used to identify the request. If the server receives two
+            ``CreateClusterRequest`` requests with the same id, then the second request will be ignored and
+            the first ``google.longrunning.Operation`` created and stored in the backend is returned.
+        :type request_id: str
+        :param retry: A retry object used to retry requests. If ``None`` is specified, requests will not be
+            retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: The amount of time, in seconds, to wait for the request to complete. Note that if
+            ``retry`` is specified, the timeout applies to each individual attempt.
+        :type timeout: float
+        :param metadata: Additional metadata that is provided to the method.
+        :type metadata: Sequence[Tuple[str, str]]
+        """
+        # Dataproc labels must conform to the following regex:
+        # [a-z]([-a-z0-9]*[a-z0-9])? (current airflow version string follows
+        # semantic versioning spec: x.y.z).
+        labels = labels or {}
+        labels.update({'airflow-version': 'v' + airflow_version.replace('.', '-').replace('+', '-')})
+
+        cluster = {
+            "project_id": project_id,
+            "cluster_name": cluster_name,
+            "config": cluster_config,
+            "labels": labels,
```

Comment:
```suggestion
            "labels": labels,
            "request_id": request_id,
```

Reply:
This param causes exception.
```
Traceback (most recent call last):
  File "/opt/airflow/airflow/models/taskinstance.py", line 1086, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/opt/airflow/airflow/models/taskinstance.py", line 1260, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/opt/airflow/airflow/models/taskinstance.py", line 1300, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/airflow/providers/google/cloud/operators/dataproc.py", line 605, in execute
    cluster = self._create_cluster(hook)
  File "/opt/airflow/airflow/providers/google/cloud/operators/dataproc.py", line 541, in _create_cluster
    metadata=self.metadata,
  File "/opt/airflow/airflow/providers/google/common/hooks/base_google.py", line 383, in inner_wrapper
    return func(self, *args, **kwargs)
  File "/opt/airflow/airflow/providers/google/cloud/hooks/new_dataproc.py", line 298, in create_cluster
    metadata=metadata,
  File "/usr/local/lib/python3.7/site-packages/google/cloud/dataproc_v1/services/cluster_controller/client.py", line 406, in create_cluster
    request.cluster = cluster
  File "/usr/local/lib/python3.7/site-packages/proto/message.py", line 579, in __setattr__
    pb_value = marshal.to_proto(pb_type, value)
  File "/usr/local/lib/python3.7/site-packages/proto/marshal/marshal.py", line 205, in to_proto
    pb_value = rule.to_proto(value)
  File "/usr/local/lib/python3.7/site-packages/proto/marshal/rules/message.py", line 32, in to_proto
    return self._descriptor(**value)
ValueError: Protocol message Cluster has no "request_id" field.
```

Reply:
There is `request` parameter which allows to pass `SubmitJobRequest` which allows users to pass `request_id`

Reply:
Also there's a new tooling around this `SubmitJobRequest` which may allow us to improve the idempotency